ThreeSum.py

```diff
@@ -0,0 +1,31 @@
+'''
+3Sum
+Given an array S of n integers, are there elements a, b, c in S such that a + b + c = 0?
+Find all unique triplets in the array which gives the sum of zero.
+Note: The solution set must not contain duplicate triplets.
+'''
+
+class Solution():
+    def threeSum(self, arr):
+        if len(arr) < 3 or arr == []:
+            return []
+        arrOfArr = []
+        for i in range(0, len(arr)):
+            if (i+1) < len(arr):
+                if (-(arr[i] + arr[i+1])) in arr:
+                    subArr=[arr[i], arr[i+1], -(arr[i] + arr[i+1])]
```

Comment:
Hi!  For this if statement ->   _**if (-(arr[i] + arr[i+1])) in arr:**_  , wouldn't it be only checking the sum of adjacent elements that can add another number to 0? i.e [-1,0,1,-3,4,3] -> [-1,0,1] will be part of the solution. But, [0,-3,3] will may not be part of the solution. Since the elements in the original array are not adjacent to each other.